src/logger.js

```diff
@@ -1,11 +1,37 @@
 let PRODUCTION = false
+let TEST = false
+let listeners = {};
 
 try {
   PRODUCTION = process.env.NODE_ENV === 'production'
 } catch (e) { }
 
+try {
+  TEST = process.env.NODE_ENV === 'test'
+} catch (e) { }
+
+export function listen(key, listener){
+  if(!TEST && !PRODUCTION && console && typeof console.error === 'function'){
+    console.error('listen() is only for use in test')
+  }
+  else {
+    listeners[key] = listener
+  }
+}
+export function clearListeners(){
+  if(!TEST && !PRODUCTION && console && typeof console.error === 'function'){
+    console.error('clearListeners() is only for use in test')
+  }
+  else {
+    listeners = {};
+  }
+}
+
 export default function (...args) {
-  if (!PRODUCTION && console && typeof console.error === 'function') {
+  if(TEST && Object.keys(listeners).length > 0) {
+    Object.keys(listeners).forEach(function(key){listeners[key](...args)});
+  }
+  else if (!PRODUCTION && console && typeof console.error === 'function') {
```

Comment:
I wonder if we could accomplish this with Jest mocks.

https://facebook.github.io/jest/docs/en/manual-mocks.html#content

Related: https://stackoverflow.com/questions/41223963/jest-how-to-mock-console-when-it-is-used-by-a-third-party-library

Reply:
Yes, that's actually what I was looking for. But at first glance, it didn't seem like they were going to serve the purpose I had in mind. On a closer reading, yes, the mocks approach looks better. I'll tweak.